### PR TITLE
migrate to api.search.io

### PR DESCRIPTION
**Pull request by benhinchley**
This change migrates to [`api.search.io`](https://dev.search.io/docs/api/) from `jsonapi.sajari.net`.

The major things that have changed are:

- `Client` was rewritten.
- `Client.query` returns the API response directly (we no longer modify the response in the client).
- `QueryPipeline` was removed, including the `Client.pipeline` method.
- `Session` and all implementations of it were removed. `SearchIOAnalytics` should be used instead.
- `PosNegLocalStorageManager` was removed. `SearchIOAnalytics` should be used instead.
- `Values` was removed.
- `Filter` was removed.
- `EventEmitter` was removed.

ENG-3242

### CODE REVIEW

**Comment by wwalser** [comment]
Just saw this. Will try to have a look tomorrow.

**Comment by justinkoke** [comment]
@wwalser and @JasonBerry is this PR going stale? Or is it still on the cards?

**Comment by wwalser** [comment]
It'll ship. Just making sure our ducks are in a row for doing a major
version bump.

**Comment by benhinchley** [comment]
@wwalser If the scope of this is too large, happy to reduce it down so it is just the client change, and not removing the deprecated/unused code.

File: .changeset/small-ravens-trade.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+This change migrates to [`api.search.io`](https://dev.search.io/docs/api/)
+from `jsonapi.sajari.net`.
+
+The major things that have changed are:
+
+- `Client` was rewritten.
+- `Client.query` returns the API response directly (we no longer modify the
+  response in the client).
```
[the remark this answers is not in the document]
Comment (benhinchley) [reply .changeset/small-ravens-trade.md:12]: Yeah okay fair enough, leave this with me I'll work on reducing the scope of this PR to just the client change (and the pieces that effects)

**Pull request closed without merging**